### PR TITLE
Add Fullscreen option

### PR DESCRIPTION
**Pull request by neopilou**
Fixes: https://github.com/home-assistant/home-assistant-android/issues/230
Add Fullscreen option in app settings :

Screenshot_20200125-113348_Home Assistant

### CODE REVIEW

**Comment by JBassett** [comment]
I'll validate on a couple devices tonight just to be sure then merge it.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsPresenterImpl.kt [hunk 1]
```diff
@@ -39,6 +40,7 @@ class SettingsPresenterImpl @Inject constructor(
             when (key) {
                 "location_zone" -> integrationUseCase.setZoneTrackingEnabled(value)
                 "location_background" -> integrationUseCase.setBackgroundTrackingEnabled(value)
+                "fullscreen" -> integrationUseCase.setFullScreenEnabled(value)
                 else -> throw Exception()
             }
             settingsView.onLocationSettingChanged()
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/settings/SettingsPresenterImpl.kt]: This is being called when you toggle full screen and it is causing a prompt for location settings.  The `settingsView.onLocationSettingChanged()` should only be called when a location settings has changed, not whenever a boolean setting changed.

File: app/src/main/res/xml/preferences.xml [hunk 2]
```diff
@@ -37,6 +37,13 @@
             android:title="@string/pref_location_background_title"
             android:summary="@string/pref_location_background_summary"/>
     </PreferenceCategory>
+    <PreferenceCategory
+        android:title="@string/other_settings">
+        <SwitchPreference
```
Comment (JBassett) [inline app/src/main/res/xml/preferences.xml]: I saw the icon added in source but you aren't referencing it here?

**Comment by JBassett** [comment]
The code is functional, just need to fix my comments before we can merge.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsPresenterImpl.kt (hunk 1 above)
Comment (neopilou) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsPresenterImpl.kt]: Add condition for settingsView.onLocationSettingChanged() call

File: app/src/main/res/xml/preferences.xml (hunk 2 above)
Comment (neopilou) [reply app/src/main/res/xml/preferences.xml]: Fullscreen icon reference added

**Pull request merged**